doc/users_guide.rst

```diff
@@ -0,0 +1,306 @@
+
+
+Transitioning from LISA legacy
+==============================
+
+A big refactoring effort was started in mid 2018, which produced a lot of
+(much needed) changes. If you are used to using LISA before the refactoring came
+into place, this guide is for you.
+
+Project structure
++++++++++++++++++
+
+* ``lisa/libs/utils`` is now ``lisa/lisa/``. ``lisa/libs/wlgen`` has also been
+  moved to that location.
+* devlib, trappy and bart are now under ``lisa/external/``. Git subtrees are
+  now used instead of submodules.
+* ``lisa/tests/lisa`` is now ``lisa/tests``
+* ``lisa/tests/eas`` is now ``lisa/lisa/tests``
+
+Imports
++++++++
+
+LISA legacy used relative imports with a bit of dark magic to hold everything
+together. Say you want to import LISA the :class:`lisa.trace.Trace` class found
+in ``lisa/trace.py``, previously you would do it like so::
+
+  from trace import Trace
+
+However, relative imports are dangerous - did you know :class:`trace.Trace` exists
+in Python's standard library? This means that the above import would import done
+in a non-LISA environment would import something completely different!
+
+.. tip::
+
+  This can help you figure out what you are really importing:
+
+    >>> import trace
+    >>> print(trace.__file__)
+    /usr/lib/python3.5/trace.py
+
+We now enforce the use of absolute imports, which look like this::
+
+  from lisa.trace import Trace
+
+.. warning::
+
+  Do make sure you haven't kept some ``PYTHONPATH`` tweaking in your ``.bashrc``
+  that would lead to an older LISA/devlib/trappy being imported.
+
+TestEnv
++++++++
+
+Configuration
+-------------
+
+Creating a :class:`lisa.env.TestEnv` used to look like this::
+
+    target_conf = {
+	# Define the kind of target platform to use for the experiments
+	"platform"    : 'linux',
+
+	# Preload settings for a specific target
+	"board"       : 'juno',  # juno - JUNO board with mainline hwmon
+
+	# Define devlib module to load
+	"modules"     : [
+	    'bl',           # enable big.LITTLE support
+	    'cpufreq'       # enable CPUFreq support
+	],
+
+	"host"        : '192.168.0.1',
+	"username"    : 'root',
+	"password"    : 'root',
+
+	"rtapp-calib" : {
+	    '0': 361, '1': 138, '2': 138, '3': 352, '4': 360, '5': 353
+	}
+    }
+
+    te = TestEnv(target_conf)
+
+We now have a dedicated class for the ``target_conf``, see :class:`lisa.env.TargetConf`.
+The most notable changes are as follows (see the doc for details):
+
+* ``"platform"`` is now ``"kind"``
+* ``"board"`` used to load some target-specific settings, which we got rid of.
+  The closest thing to it is ``"name"`` which is just a pretty-printing name and
+  has no extra impact.
+* You don't have to specify devlib modules to load anymore. All (loadable)
+  modules are now loaded. If you find some module too slow to load, you can
+  specify a list of modules to exclude.
+* LISA used to have ``target.config`` JSON file at its root. Its equivalent is
+  now ``target_conf.yml``, which is in YAML.
+
```

Comment:
should also mention `PlatformInfo`, since it is the place to modify for `rtapp` calib values. While at it, we can also mention `!var package.module.global` YAML tag we have to reference Python objects directly. This unifies the long-standing JSON-vs-module issue, e.g. `lisa/platforms/juno_r0.yml`